subprojects/shipkit/src/test/groovy/org/shipkit/internal/gradle/git/GitHubUrlBuilderTest.groovy

```diff
@@ -14,8 +14,28 @@ class GitHubUrlBuilderTest extends Specification {
         GitHubUrlBuilder.getGitHubUrl("org/repo", conf) == "https://user:token@github.com/org/repo.git"
     }
 
+    def "should return Custom GH url whith auth"() {
+        def conf = new ShipkitConfiguration()
+        conf.gitHub.writeAuthToken = "token"
+        conf.gitHub.writeAuthUser = "user"
+        conf.gitHub.url = "https://github.enterprise.com:123456"
+
+        expect:
+        GitHubUrlBuilder.getGitHubUrl("org/repo", conf) == "https://user:token@github.enterprise.com:123456/org/repo.git"
+    }
+
+    def "should return Custom GH url whith auth, even with trailing /"() {
+        def conf = new ShipkitConfiguration()
+        conf.gitHub.writeAuthToken = "token"
+        conf.gitHub.writeAuthUser = "user"
+        conf.gitHub.url = "http://192.168.0.199/"
+
+        expect:
+        GitHubUrlBuilder.getGitHubUrl("org/repo", conf) == "http://user:token@192.168.0.199/org/repo.git"
+    }
+
     def "should return GH url without auth by default"() {
```

Comment:
Is this the requirement that is tested using the null config param you mean? If so I agree, it should not be used since there is a default in the config, just like you say. I'll gladly remove it, was not necessary for my purpose and dondt wanna change anything else, must say I just starting to understand how this software works, I don't know if it's needed for something else.

Reply:
Sorry for being vague :)

Let's always use the value from ```shipkit.gitHub.url``` and avoid hardcoding the url. The test name "should return GH url without auth by default" is not correct. Instead we want something like that: "should use correct GH url when auth token is null".

Thank you for looking into it!